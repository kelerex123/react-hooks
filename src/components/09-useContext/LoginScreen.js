import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {user, setUser} = useContext(UserContext)

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />

            <button className='btn btn-primary' onClick={ () => setUser({
                id: 4568686,
                name:'Super'
            })}>
                Login
            </button>
        </div>
    )
}
