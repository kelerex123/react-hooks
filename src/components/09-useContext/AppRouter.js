import React from 'react'
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom'
import {AboutScreen} from './AboutScreen'
import {LoginScreen} from './LoginScreen'
import {HomeScreen} from './HomeScreen'
import { NavBar } from './NavBar'

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <div>
                <NavBar />


                <div className='container'>
                    <Routes>
                        <Route path='/' element={<HomeScreen />}/>
                        <Route path='/about' element={<AboutScreen />}/>
                        <Route path='/login' element={<LoginScreen />}/>
                        <Route path='*' element={<Navigate replace to='/' />}/>

                    </Routes>
                </div>
            </div>
        </BrowserRouter>
    )
}
