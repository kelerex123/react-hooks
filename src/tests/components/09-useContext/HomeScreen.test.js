import {shallow, mount} from 'enzyme'
import { HomeScreen } from '../../../components/09-useContext/HomeScreen'
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas con <HomeScreen />', () => {

    const user = {
        name: 'Bryan',
        email: 'correo@gmail.com'
    }
    
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    })

})