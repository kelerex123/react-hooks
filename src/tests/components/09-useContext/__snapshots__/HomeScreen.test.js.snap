// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <HomeScreen /> debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Bryan",
   "email": "correo@gmail.com"
}
  </pre>
</div>
`;
